### PR TITLE
fix(angular): return to correct tab outside tab context

### PR DESCRIPTION
**Pull request by Zapominacz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features) 
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Currently, after leaving ion-tabs context (for example pushing route that is defined outside the tabs routing paths) the view hierarchy is preserved. However, after moving back using ion-back-button or swiping back, after transition completed - we are routed back to the default tab.

Issue Number: #19801

## What is the new behavior?

For now, we are returning to the previous tab (or whether route was activated inside).

To achieve that, I'm searching the view that is the leaf of the routing hierarchy, branched from the view on the stack (in this case - tabs), and using its url (or context) to resolve the correct destination (consistent with the visible underlying view).
In other words, I'm looking for the deepest view I can find inside the view we are backing to.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Tested on reproduction present in #23218 and in my project. Any suggestions are welcome.

### CODE REVIEW

**Review by ranemihir — APPROVED** [review]

**Review by mrtmarkov150 — APPROVED** [review]

**Review by mrtmarkov150 — CHANGES_REQUESTED** [review]
F

**Comment by Zapominacz** [comment]
I am removing that pull request ass 5.6.6 fixes that issue
